lib/include/public/LogManagerBase.hpp

```diff
@@ -273,6 +299,9 @@ namespace MAT_NS_BEGIN
         /// </summary>
         static status_t FlushAndTeardown()
         {
+            if (isGone())
+                return STATUS_EALREADY;
```

Comment:
...the process is dying at this point. No need to teardown. Trying to do teardown is detrimental since corresponding config object is already dead, and so we would have tripped on it. Maybe the `stateLock` below (also uses global static) should also have similar global state check.